### ISSUE TITLE
relationship of the mvp package to the spray and freealg packages

### ISSUE THREAD

**Issue by RobinHankin** [RobinHankin/mvp#47]
It is a pain to maintain the dependency on the `spray` package, which was originally placed to make the timings section in the vignette work.  But I have discovered that the supposed  efficiency advantages of the `mvp` package are not a clear-cut as I had thought.  I want to remove the dependency on `spray`.

**Comment by RobinHankin** [comment]
The best idea I have had so far is to create a non-required vignette in the `inst/` directory which defines the coercion functions [currently `spray_to_mvp()` and `mvp_to_spray()`] and presents some timings.

and discuss the connection there.

**Comment by RobinHankin** [comment]
Similar comments for the `freealg` package; we have `freealg_to_mvp()` to consider.   I will edit the title of this issue.  My view now is that packages should be as self-contained as possible.

**Comment by RobinHankin** [comment]
Look at the current definition of `as.mvp()`:

```
`as.mvp` <- function(x,...){
  if(is.mvp(x)){
    return(x)
  } else if(is.mpoly(x)){
    return(mpoly_to_mvp(x))
  } else if(inherits(x,"spray")){
    return(spray_to_mvp(x,...))
  } else if(inherits(x,"freealg")){ # free algebra
    return(freealg_to_mvp(x))
  } else if(is.character(x)){
    return(mpoly_to_mvp(mp(x)))
  } else if(is.list(x)){
    return(mvp(vars=x$names,powers=x$power,coeffs=x$coeffs))
  } else if(is.numeric(x)){
    return(numeric_to_mvp(x))
  } else {
    stop("not recognised")
  }
}
```

Function `as.mvp()` needs to import `freealg_to_mvp()` and `spray_to_mvp()` from somewhere.  Function `as.mvp()` should be an `S3` generic to avoid requiring the `mvp` and `freealg` packages (and then define methods in those packages).   Either that or simply remove the function's ability with these objects.

**Comment by RobinHankin** [comment]
fixed in 6e508a6
